src/lib/components/api.js

```diff
@@ -0,0 +1,164 @@
+const githubOwner = "SSWConsulting";
+const githubRepo = "SSW.Rules.Content";
+const apiBaseUrl = "https://api.github.com/graphql";
+
+export async function fetchPullRequests(numberOfRules, author, token) {
+  const pullRequests = await requestPullRequests(
+    numberOfRules,
+    author,
+    token
+  );
+  pullRequests.sort((a, b) => new Date(b.mergedAt) - new Date(a.mergedAt));
+  return pullRequests;
+}
+
+export async function fetchFileContents(filesToRetrieve, numberOfRules, token) {
+  var fileContents = await requestMultipleFileContents(
+    filesToRetrieve,
+    numberOfRules,
+    token
+  );
+
+  var counter = 0;
+
+  while (
+    fileContents.filter((x) => x !== null).length < numberOfRules
+  ) {
+    var extraFile = await requestSingleFileContents(
+      filesToRetrieve[counter + numberOfRules].path,
+      token
+    );
+    fileContents = [...fileContents, extraFile];
```

Comment:
What is the use case here? To me it looks like sometimes the API returns null file content when we query it but I am confused as to why.